### PR TITLE
Add containerd2 package

### PR DESCRIPTION
**Pull request by liunan-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
As containerd 2.0 GA recently, we want to provide containerd 2.0 in azure linux 3.0 to enhance our distro. Currently we wanted to keep the current containerd version in the existing package and offer container 2.0 in a new package so that users can try and test as their needs. The original PR #11029 is split into two PRs, one #11383 is to upgrade runc to 1.2.2 and libseccomp to 2.5.5 as the dependencies of containerd 2.0.0, and this PR is based on it and adds containerd 2.0.0 package.

###### Change Log  

- Created a standalone package for containerd 2.0.0

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Buddy build id: [692899](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=692899&view=results)
- AKS Tests passed:
  - Conformance tests: [Pipelines - Run 3.0_20241108_671573](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=671573&view=results)
  - Performance tests: [Pipelines - Run 20241111.4](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=673679&view=results)
  - Pre-release-tests: [Pipelines - Run 20241107.2](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=671132&view=results)
  - E2e tests: [Pipelines - Run 20241109.1 logs](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=107775151&view=logs&j=e4db8e77-e27d-5659-0ddb-014244fdf1b3&t=1d4f3824-138a-5ea4-ea39-cffb8c23d243)
- BVT: [Pipelines - Run 3.0.20241204-liunan-688614](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=688837&view=results) passed
- DockerTestSuite in Lisa test passed: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=693959&view=logs&j=528a33a8-464e-565a-7855-df1244039b47&t=6d31a32c-15ff-58cb-365f-6bf2dc6af640&l=1893
- Perf analysis across multiple containerd and runc versions: https://microsoft-my.sharepoint.com/:x:/p/liunan/EWt7afiVvANAq7RF8-9n2IcB57Dtr14j1mpHeIvW2qywBQ?e=D7ZEib

### CODE REVIEW

File: SPECS/containerd2/containerd2.spec [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+%{_bindir}/*
+%{_mandir}/*
+%config(noreplace) %{_unitdir}/containerd.service
+%config(noreplace) %{_sysconfdir}/containerd/config.toml
+%dir /opt/containerd
+%dir /opt/containerd/bin
+%dir /opt/containerd/lib
+
+%changelog
+* Wed Dec 11 2024 Nan Liu <liunan@microsoft.com> - 2.0.0-1
```
Comment (ms-mahuber) [inline SPECS/containerd2/containerd2.spec:83]: This looks like a newly introduced package - why the long changelog?

File: SPECS/containerd2/containerd.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#  level = "info"
+
+version = 2
+[plugins]
+  [plugins."io.containerd.grpc.v1.cri"]
+   [plugins."io.containerd.grpc.v1.cri".containerd]
+      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+          runtime_type = "io.containerd.runc.v2"
+          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
```
Comment (ms-mahuber) [inline SPECS/containerd2/containerd.toml:24]: Did we make thoughts wrt. this?

https://github.com/Azure/AgentBaker/commit/440a39716605d4400b9fbf365c4ad248e0059028

File: SPECS/containerd2/containerd2.spec (hunk 1 above)
Comment (liunan-ms) [reply SPECS/containerd2/containerd2.spec:83]: Removed the long changelog.

File: SPECS/containerd2/containerd.toml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#root = "/var/lib/containerd"
```
Comment (ms-mahuber) [inline SPECS/containerd2/containerd.toml:1]: Why do we need the commented stuff? :) Is this only ported from the prior SPEC or is this actively being utilized in some cases?

File: SPECS/containerd2/containerd2.spec [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+Vendor: Microsoft Corporation
+Distribution: Azure Linux
+
+Source0: https://github.com/containerd/containerd/archive/v%{version}.tar.gz#/%{upstream_name}-%{version}.tar.gz
+Source1: containerd.service
+Source2: containerd.toml
+
+%{?systemd_requires}
+
+BuildRequires: git
```
Comment (ms-mahuber) [inline SPECS/containerd2/containerd2.spec:21]: are we sure? this suggests to me that there may be an online dependency but we build entirely offline. For what step is git required?

File: SPECS/containerd2/containerd.toml (hunk 3 above)
Comment (liunan-ms) [reply SPECS/containerd2/containerd.toml:1]: This commented part is from the existing containerd package, we can remove them for containerd2 package.

File: SPECS/containerd2/containerd2.spec (hunk 4 above)
Comment (liunan-ms) [reply SPECS/containerd2/containerd2.spec]: Good catch, I removed git which is not needed, validated by buddybuild [693469](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=693469&view=results).

**Review by ms-mahuber — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**